cmd/jujud/agent/agent.go

```diff
@@ -164,7 +164,8 @@ func GetJujuVersion(machineAgent string, dataDir string) (version.Number, error)
 	return config.UpgradedToVersion(), nil
 }
 
-func dependencyEngineConfig() dependency.EngineConfig {
+// DependencyEngineConfig returns a dependency engine config.
+func DependencyEngineConfig() dependency.EngineConfig {
```

Comment:
I'm really hesitant to use this method from this package. The agent package has bootstrap in it, which means it pulls in all the provider code and state. Perhaps I'm overly paranoid at the skill of the go linker to exclude everything, but we could perhaps be bringing in init functions we don't want, and I guess any global variables in those packages.

I think perhaps it would be better to move this function into `cmd/jujud/agent/engine`.

Remembering that our primary focus here is to minimize dependencies.

I think we should move the IsFatal method (and type) along with MoreImportantError into a new package like `cmd/jujud/agent/errors`, and move the `upgrader.UpgradeReadyError` into that package too. We want to avoid reaching into workers to get error types.

Reply:
good point, I am moving all those things reused for both jujud and k8sagent to separate packages.